src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final byte RESIZE_CAPACITY = 2;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeTable();
+        Node<K, V> newNode = new Node<>(key, value);
+        Node<K, V> node = table[getIndex(key)];
+        int index = getIndex(key);
+        while (node != null) {
+            if (node.key == key || node.key != null && node.key.equals(key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = newNode;
+                size++;
+                return;
+            }
```

Comment:
this logic can be performed outside of the loop

Reply:
we cant do it outside of the loop because we need to check all options.

Reply:
we can enter the second 'if' block only during the last loop iteration
Alternatively, we can save the last node and then access it from outside the loop.